### PR TITLE
Fixes `Brave Ads has arrived!` notification in BAT icon is shown after browser restart on clean installs

### PR DESCRIPTION
**Pull request by tmancey**
Fixes https://github.com/brave/brave-browser/issues/7133

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

See  https://github.com/brave/brave-browser/issues/7133

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_ads/browser/ads_service_impl.cc [hunk 1]
```diff
@@ -1817,7 +1817,13 @@ void AdsServiceImpl::OnPrefsChanged(
...
       pref == brave_rewards::prefs::kBraveRewardsEnabled) {
     if (IsEnabled()) {
-      RemoveOnboarding();
+#if !defined(OS_ANDROID)
+      if (first_run::IsChromeFirstRun()) {
+        SetBooleanPref(prefs::kShouldShowOnboarding, false);
+      } else {
+        RemoveOnboarding();
+      }
+#endif
```
Comment (masparrow) [inline components/brave_ads/browser/ads_service_impl.cc]: As the original code (presumably) affected Android as well, and now this is excluded - what _should_ / does happen on Android now?

Comment (tmancey) [reply components/brave_ads/browser/ads_service_impl.cc]: For Android the original code just exited inside `RemoveOnboarding` as onboarding is handled differently for Android. There is a ticket to refactor Android to use the same business logic flags for onboarding

**Comment by tmancey** [comment]
CI failed for unrelated reasons due to network audit

File: components/brave_ads/browser/ads_service_impl.cc (hunk 1 above)
Comment (NejcZdovc) [reply components/brave_ads/browser/ads_service_impl.cc]: but now `RemoveOnboarding` is not triggered as the whole thing is wrapped for not android

Comment (tmancey) [reply components/brave_ads/browser/ads_service_impl.cc]: RemoveOnboarding does not need to be triggered for Android as they handle their own onboarding at this time

Comment (masparrow) [reply components/brave_ads/browser/ads_service_impl.cc]: @tmancey Provided Android will never rely on `RemoveOnboarding` being called in the future (as now it won't), I'm happy.

**Comment by tmancey** [comment]
Unrelated CI failure

**Pull request merged**